src/getPublicKey.c

```diff
@@ -263,43 +264,42 @@ void handleGetPublicKey(uint8_t p1,
                         uint16_t dataLength,
                         volatile unsigned int *flags,
                         volatile unsigned int *tx) {
-    // Sanity-check the command parameters.
-    if ((p2 != P2_DISPLAY_ADDRESS) && (p2 != P2_DISPLAY_PUBKEY)) {
-        // Although THROW is technically a general-purpose exception
-        // mechanism, within a command handler it is basically just a
-        // convenient way of bailing out early and sending an error code to
-        // the computer. The exception will be caught by zil_main, which
-        // appends the code to the response APDU and sends it, much like
-        // io_exchange_with_code. THROW should not be called from
-        // preprocessors or button handlers.
-        THROW(SW_INVALID_PARAM);
-    }
-
     // Read the key index from dataBuffer and set the genAddr flag according
     // to p2.
     ctx->keyIndex = U4LE(dataBuffer, 0);
-    ctx->genAddr = (p2 == P2_DISPLAY_ADDRESS);
+    ctx->genAddr = ((p2 & P2_DISPLAY_ADDRESS) != 0);
 
-    // Prepare the approval screen, filling in the header and body text.
-    if (ctx->genAddr) {
-        os_memmove(ctx->typeStr, "Generate Address", 17);
-    }
-    else {
-        os_memmove(ctx->typeStr, "Generate Public", 16);
+    // Checking whether we should generate the address / public key without verification
+    if((p2 & P2_NO_VERIFY) != 0)
```

Comment:
I will do that on Monday.

Reply:
I've added a  test under `tests/functional/test_pubkey_cmd.py` but I have been unable to confirm that it actually works so far as I run into problems when running the test suite. I get the following message:

```
[2023-01-16 08:46:37,135][INFO] ragger.logger - Starting SpeculosBackend stream
qt.qpa.xcb: could not connect to display
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl, xcb.

[*] speculos launcher revision:
[*] using SDK version 2.1 on nanos
[*] loading CXLIB from "/usr/local/lib/python3.8/dist-packages/speculos/cxlib/nanos-cx-2.1.elf"
[*] patching svc instruction at 0x126358
[*] patching svc instruction at 0x4000385c
[*] patching svc instruction at 0x4000386a
------------------------------------------------------------------- Captured log setup -------------------------------------------------------------------
INFO     ragger.logger:speculos.py:84 Starting SpeculosBackend stream
INFO     speculos-client:client.py:185 starting speculos with command: /usr/bin/python3 -m speculos --display qt --model nanos --sdk 2.1 /ledger-app/app/tests/elfs/zilliqa_nanos.elf
```
After this, `qemu` seems to exit and all tests fail with a message that it could not connect to the emulator. Have you seen this error message before?

Reply:
Test added and passing on my local machine.